### ISSUE TITLE
Remove defaultHighlight prop

### ISSUE THREAD

**Issue by oliviertassinari** [mui-org/material-ui-pickers#1677]
Let's say a user open the date picker on a desktop with value = null. He will see the current day with the same style as a selected day, unlike what the user might think, today isn't selected.
With hindsight, I don't see the use case for the `defaultHighlight` prop. I propose we remove it.

**Comment by Philipp91** [comment]
I have a use case for it: I pass "<today> <midnight>" to a `DateTimePicker` such that this becomes the default date when the user just opens and closes a picker whose value was previously `null`. This is different from the default behavior, which would fill in "<today> <current time>" instead (simply `new Date()`). In my application, midnight means that there's no time associated with the value (00:00 is just a placeholder and I modify things to not display the time at all in this case).

I can imagine other use cases in other applications for suggesting a different date than "right now" as the default value without filling it into the text box or making it the current "value" of the picker. I.e. it should only matter when the picker is opened with current value `null`. For example, I can imagine applications that suggest the next weekday, the first of the next month, etc.

**Comment by oliviertassinari** [comment]
@Philipp91 Humm, this sounds very much like the `autoSelect` and `autoHighlight` props of the [Autocomplete](https://material-ui.com/components/autocomplete/#playground). I don't think that that we need to provide the behavior with a prop with the date pickers. Instead, you could listen to the onClose state (to implement autoSelect), as well as control the selectedDays state to implement autoHighlight.

**Comment by Philipp91** [comment]
Do you mean `onOpen`? I guess that would work.

**Comment by Philipp91** [comment]
Something like `onOpen={() => { if (!value) onChange(myDesiredDefaultDate); }}` sort of works, but it misbehaves on mobile because the value stays set even if the user clicks "Cancel", whereas the correct behavior would be to leave the field unchanged (i.e. still `null`) in case of cancelling. WDYT?

**Comment by Philipp91** [comment]
Another reason why I currently need `defaultHighlight`: I want my dates to have the time `00:00:00`, but the default `DatePicker` will return the picked date with the current time because it uses `new Date()` and then just changes the date portion of the value. I currently work around that by feeding `defaultHighlight={getMidnightToday()}`.

**Comment by oliviertassinari** [comment]
@Philipp91 This sounds like another reason to remove `defaultHighlight`. We discuss this problem in #1526.

**Comment by Philipp91** [comment]
Okay. It's a bit confusing that the library internally calls `new Date()`, but you're right, I can do without `defaultHighlight` and just post-process the values I'm getting out of `onChange`.

**Comment by oliviertassinari** [comment]
@Philipp91 Yes, my concern is that relying on `defaultHighlight` is brittle.
